packages/runtime-core/src/renderer.ts

```diff
@@ -2386,24 +2387,6 @@ function baseCreateRenderer(
       instance.isUnmounted = true
     }, parentSuspense)
 
-    // A component with async dep inside a pending suspense is unmounted before
-    // its async dep resolves. This should remove the dep from the suspense, and
-    // cause the suspense to resolve immediately if that was the last dep.
-    if (
-      __FEATURE_SUSPENSE__ &&
-      parentSuspense &&
-      parentSuspense.pendingBranch &&
-      !parentSuspense.isUnmounted &&
-      instance.asyncDep &&
-      !instance.asyncResolved &&
-      instance.suspenseId === parentSuspense.pendingId
-    ) {
-      parentSuspense.deps--
```

Comment:
The logic here will no longer be executed because when the component is unmounting, `parentSuspense.pendingId` has already changed.
see https://github.com/vuejs/core/blob/163b3651d174321911648a164052effa9249a2aa/packages/runtime-core/src/components/Suspense.ts#L278